apps/designer/app/canvas/canvas.tsx

```diff
@@ -73,12 +79,21 @@ const useElementsTree = () => {
       return;
     }
 
-    return createElementsTree({
-      sandbox: true,
-      instance: rootInstance,
-      Component: WrapperComponentDev,
-      onChangeChildren,
-    });
+    return (
+      <ReactSdkProvider
```

Comment:
This makes canvas fully coupled to react SDK ... how is this supposed to work some day with a different framework?

Reply:
It is coupled anyway. With other framework we'll use another way to inject stores.

Reply:
how is it coupled anyway? we are trying to remove every coupling in every step we do ...
I think react provider that has to be provided is not the ideal way of doing it, SDK should provide something agnostic so we don't have to assume on canvas that its already using react.

there is a bunch of options

Reply:
Renderer is full react. It won't be reused in any way. We will reimplement it in other frameworks.

Reply:
I can put provider inside createElementsTree if you want

Reply:
Anything works for me that doesn't create more coupling with react in SDK. Currently you require SDK to provide this provider

Reply:
done